stor/dx.py

```diff
@@ -436,17 +666,19 @@ def canonical_project(self):
         """
         if utils.is_valid_dxid(self.project, 'project'):
             return self.project
-        else:
-            try:
-                proj_dict = dxpy.find_one_project(
-                    name=self.project, level='VIEW', zero_ok=True, more_ok=False)
-                if proj_dict is None:
-                    raise ProjectNotFoundError('No projects were found with given name ({})'
-                                               .format(self.project))
-                return proj_dict['id']
-            except DXError as e:
-                raise DuplicateProjectError('Duplicate projects were found with given name ({})'
-                                            .format(self.project), e)
+
+        try:
+            proj_dict = dxpy.find_one_project(
```

Comment:
I think we'd previously discussed not assuming output from this and instead just trying to find projects with a limit of 2 then raising exceptions based on that?

Reply:
Yep, please look at my reply here:  https://github.com/anujkumar93/stor/pull/2#discussion_r216771573